WinUIGallery/Controls/HomePageHeaderImage.xaml.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using System.Numerics;
+using Microsoft.Graphics.Canvas.Effects;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Hosting;
+using Microsoft.UI.Xaml.Media.Animation;
+using Microsoft.UI.Composition;
+using Microsoft.UI;
+using CommunityToolkit.WinUI.UI.Animations;
+using Windows.UI;
+using System;
+
+// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
+
+namespace AppUIBasics.Controls
+{
+    // ATTRIBUTION: @RykenApps
+    public sealed partial class HomePageHeaderImage : UserControl
+    {
+        private Compositor _compositor;
+        private CompositionLinearGradientBrush _imageGridBottomGradientBrush;
+        private CompositionEffectBrush _imageGridEffectBrush;
+        private ExpressionAnimation _imageGridSizeAnimation;
+        private ExpressionAnimation _bottomGradientStartPointAnimation;
+        private SpriteVisual _imageGridSpriteVisual;
+        private CompositionSurfaceBrush _imageGridSurfaceBrush;
+        private Visual _imageGridVisual;
+        private CompositionVisualSurface _imageGridVisualSurface;
```

Comment:
This seems quite a lot, especially the fact that some of those are animations objects. Is that really the best way to solve the problem of fading out an image?

Reply:
Ah, good catch - I forgot to push the zoom in animation on app load.

For the gradient: this is actually trickier than expected and I couldn't find a lot online.. having the transparency-fade in the image wouldn't work on resizing the window, using a LinearGradientBrush wouldn't work either when Mica is enabled. This allows us to put in any image and have a nice fade effect based on Composition. But obviously open to other code suggestions :)

Reply:
It seems it's really not that trivial. I would probably not have implemented that animation due to the complexity of it but I don't have a strong opinion on that.